test/test_csvs.py

```diff
@@ -0,0 +1,291 @@
+                '_Units'              : self._get_units(),
+                '_DataLevel'          : ('','L0','L1','L2')
+               }
+
+    def _get_validation_positions(self):
+        return {'_Array ID'    : (0,8),
+                '_Platform ID' : (9,14),
+                '_Instrument'  : (18,27)
+               }
+
+    def _validate_line(self, header, flds_to_validate, line):
+        validation_coll = self._get_validation_collection()
+        validation_pos = self._get_validation_positions()
+
+        invalid = []
+        parts = line.split('|')
+        for index,part in enumerate(parts):
+            if index in flds_to_validate:
+                # the common fields validated by a collection of values
+                #  (ReferenceDesignator, _ParameterID_R/T, _Units)
+                if header[index] in validation_coll:
+                    vcoll = validation_coll[header[index]]
+                    if part not in vcoll:
+                        invalid.append(header[index] + ': [' + part + ']')
+                # the common fields validated by a portion of the ReferenceDesignator field
+                elif header[index] in validation_pos:
+                    beg,end = validation_pos[header[index]]
+                    refdesidx = header.index('ReferenceDesignator')
+                    if part != parts[refdesidx][beg:end]:
+                        invalid.append(header[index] + ': [' + part + ']')
+                # the global and gradient fields
+                elif header[index] in ('GlobalRangeMin','GlobalRangeMax','GradientTest_mindx'):
+                    if not self._is_empty_or_numeric(part):
+                        invalid.append(header[index] + ': [' + part + ']')
+                # the spike fields
+                elif header[index] in ('SpikeTest_ACC','SpikeTest_N','SpikeTest_L'):
+                    if not self._is_empty_or_numeric(part):
+                        invalid.append(header[index] + ': [' + part + ']')
+                # the stuck fields
+                elif header[index] in ('StuckValueTest_ResolutionR','StuckValueTest_NumRepeatValues'):
+                    if not self._is_empty_or_numeric(part):
+                        invalid.append(header[index] + ': [' + part + ']')
+                # the trend fields
+                elif header[index] in ('TrendTest_TimeIntLengthDays','TrendTest_PolynomialOrder','TrendTest_nstd'):
+                    if not self._is_empty_or_numeric(part):
+                        invalid.append(header[index] + ': [' + part + ']')
+
+        return invalid
+
+    # ###########################################
+    # verify the files needed for the tests exist
+    # ###########################################
+
+    # the PSVs are built from the CSVs in setuptest.sh to make them pipe-separated files
+    # due to the "local" file using commas inside the fields of a handful of records
+
+    def test_globalFileExists(self):
```

Comment:
recommend one function to check all the qc files

also PEP8 - function names should be lowercase separated by underscores

Reply:
I think I've addressed all these now